toolkit/scripts/imggen.mk

```diff
@@ -5,7 +5,13 @@ $(call create_folder,$(IMAGEGEN_DIR))
 
 # Resources
 config_name              = $(notdir $(CONFIG_FILE:%.json=%))
-config_other_files       = $(if $(CONFIG_FILE),$(call shell_real_build_only, find $(CONFIG_BASE_DIR)))
+CONFIG_BASE_DIR          = $(dir $(CONFIG_FILE:%.json=%))
+config_other_files      := $(shell grep -E '.json|.sh' $(CONFIG_FILE)| sed 's/\"Path\"\://' | tr ", \n" " " | tr "\"" " " | xargs)
```

Comment:
I'm still leaning towards a dedicated script for this, say `./toolkit/scripts/get_config_deps.sh` or something? I feel like makefile functions are something we want to do sparingly if possible.

Also, I tried to poke at `jq` quickly, its honestly not too bad for this use case.
```bash
config_other_files_relative="$(jq -r '.SystemConfigs[]?.PackageLists[]' "${config_file}")"
config_other_files_relative="${config_other_files_relative} $(jq -r '.SystemConfigs[]?.PostInstallScripts[]?.Path' "${config_file}")"
```

The one caveat is that we haven't traditionally required `jq` as a build prerequisite. I'm thinking, especially if this lives in a shell script, that we could take both approaches. If `jq` is available we use it, otherwise fall back to the grep/sed approach.

Reply:
(I'm aware I've added more than my fair share of indecipherable makefile functions... I'm trying to convert those to scripts where it makes sense as I update them).